packages/calypso-e2e/src/lib/flows/start-site-flow.ts

```diff
@@ -49,7 +49,9 @@ export class StartSiteFlow {
 	 * @param {string} name Name for the blog.
 	 */
 	async enterBlogName( name: string ): Promise< void > {
-		await this.page.fill( selectors.blogNameInput, name );
+		await this.page.waitForLoadState( 'networkidle' );
```

Comment:
I tried a few different approaches and in the end it looks like the issue was being caused by Playwright over-enthusiastically filling the input fields prior to the page load being completed.